### DISCUSSION TITLE
Pass filenames without flags/options

### DISCUSSION THREAD

**Discussion by chrysante** [CLIUtils/CLI11#882] [General]
This will probably be trivial to answer, but I just can't figure it out. How can I parse filename or directory arguments without them being 'options'? Basically I would like to invoke my executable like this: 

`./my_app -x -y --some-other-option input-a.txt input-b.txt`

Instead of this:

`./my_app -x -y --some-other-option --file input-a.txt --file input-b.txt`

I guess it has something to do with positional arguments, but I can't figure out how to use them. Any pointers would be appreciated.

Thanks!

**Comment by phlptp** [comment]
``` C++
std::vector<std::string> files
app.add_option("files",files,"file to use")->check(CLI::ExistingFile)
/// all other options
```
this adds a positional option and puts all positional arguments in a vector

**Reply by phlptp** [reply]
adding an option with no `--` in front of the strings creates the positional arguments if you wanted to both it would be 
```c++
app.add_option("--file,files",files,"file to use")->check(CLI::ExistingFile)
```
the the positional works and you can specify with an option as well

**Reply by chrysante** [reply]
Works like a charm, thank you very much! Also for the great library :)
